assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -181,7 +210,59 @@ export const actions = {
 	),
 };
 
-export const controls = {};
+export const controls = {
+	[ POPULATE_QUEUE ]: createRegistryControl(
+		( registry ) =>
+			async ( { payload } ) => {
+				const { viewContext, groupID } = payload;
+				const { isNotificationDismissed } =
+					registry.select( CORE_NOTIFICATIONS );
+				const notifications = registry
+					.select( CORE_NOTIFICATIONS )
+					.getNotifications();
+
+				// Wait for all dismissed items to be available before filtering.
+				await registry.resolveSelect( CORE_USER ).getDismissedItems();
+
+				let potentialNotifications = Object.values( notifications )
+					.filter(
+						( notification ) => notification.groupID === groupID
+					)
+					.filter( ( notification ) =>
+						notification.viewContexts.includes( viewContext )
+					)
+					.filter( ( { isDismissible, id } ) =>
+						isDismissible ? ! isNotificationDismissed( id ) : true
+					)
```

Comment:
Can't we combine these 3 filters into one? From the theoretical point of view, it will be more efficient than having three separate filters.

Reply:
Ok, let it be it.